### PR TITLE
Fixing Felt Recipe

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Bugfixes "Felt Recipe"

#### Purpose of change
Felt making was missing several steps/tools.  This fixes that. https://www.wikihow.com/Make-Felt was what I used for reference.  I left the time at fifteen minutes as a single felt patch is pretty small.
edit: After listening to several primitive felt making podcasts and videos and taking into account Tommy's comments about don't modernize primitive recipes. This method is considered clumping felting and is not for recommended for beginners but doesn't require any additional tools. 

#### Describe the solution
Updates recipe.

#### Describe alternatives you've considered
none

#### Testing
simple change if it doesn't break our tests it's going to work.

### CODE REVIEW

File: data/json/recipes/other/materials.json [hunk 1]
```diff
@@ -209,7 +209,9 @@
     "difficulty": 1,
     "time": "15 m",
     "autolearn": true,
-    "components": [ [ [ "soap", 5 ] ], [ [ "water_clean", 3 ] ], [ [ "wool_staple", 1 ] ] ]
+    "qualities": [ { "id": "CUT", "level": 1 } ],
+    "tools": [ [ [ "bubblewrap", -1 ] ], [ [ "casserole", -1 ] ] ],
+    "components": [ [ [ "soap", 5 ] ], [ [ "water_clean", 3 ] ], [ [ "wool_staple", 1 ] ], , [ [ "bubblewrap", 1 ] ] ]
```
Comment (Maleclypse) [inline data/json/recipes/other/materials.json:214]:
```suggestion
    "components": [ [ [ "soap", 5 ] ], [ [ "water_clean", 3 ] ], [ [ "wool_staple", 1 ] ], [ [ "bubblewrap", 1 ] ] ]
```

**Comment by Tommy-os** [comment]
http://www.historyofclothing.com/textile-history/felt-and-feltmaking-history/

> Wet felting is method interlocking and compacting of wool fibers which are first drenched in soap water and then rubbed until fabric has the same thickness. Individual wool fibers are covered with small scales. When water and pressure from rubbing are applied the scales open up and the fibers tangle together. When the felt dries fibers stay tangled.

I did not find any mention of people using bubblewrap for the thousands of years that felt was produced by civilizations.

**Comment by Maleclypse** [comment]
> http://www.historyofclothing.com/textile-history/felt-and-feltmaking-history/
> 
> > Wet felting is method interlocking and compacting of wool fibers which are first drenched in soap water and then rubbed until fabric has the same thickness. Individual wool fibers are covered with small scales. When water and pressure from rubbing are applied the scales open up and the fibers tangle together. When the felt dries fibers stay tangled.
> 
> I did not find any mention of people using bubblewrap for the thousands of years that felt was produced by civilizations.
> 
> Why is bubblewrap now both mandatory and consumed upon making felt?
> 
> Why is specifically a casserole required?
> 
> What are you "cutting" in the recipe?

Cutting is making the edges of the patch at the end.  The other parts could probably be replaced with other tools.  Make edits to show what tools or tool qualities you'd like used in their place.  But soap and water aren't enough.

**Comment by Tommy-os** [comment]
http://www.woolcrafting.com/making-wool-felt.html

There are no tools required for making felt. You only need your hands, soap, water and wool. There's no cutting involved, there's no casseroles or bubblewrap involved. The recipe was fine as it was, I would recommend not using wikihow as a source for anything in the future.

**Comment by Maleclypse** [comment]
> http://www.woolcrafting.com/making-wool-felt.html
> 
> There are no tools required for making felt. You only need your hands, soap, water, wool and a hard surface (floor). There's no cutting involved, there's no casseroles or bubblewrap involved. The recipe was fine as it was, I would recommend not using wikihow as a source for anything in the future.

https://www.fiberartsy.com/basic-felting-fabric/

**Comment by Tommy-os** [comment]
I'm not sure what point you're trying to make, felt has been made for thousands of years without the use of bubblewrap or any modern tools or casserole.  Wool clumps can be easily separated by hand without cutting. The entire process is just applying a bit of pressure to condense wool fibers. You can do this with your own hands, water and soap in a manner of minutes just by using solid floor.

You can make an argument that it would be faster with certain tools, though a casserole serves absolutely no purpose in any of the guides you posted besides retaining the water so don't dirty your surroundings. Neither does a cutting implement make sense to create a patch, you shape the wool clumps with hands. Cutting comes when you're making clothing out of it, you don't cut your rags into precise shapes after you knit them.

It seems you're trying to artificially increase difficulty of this particular recipe without any basis in reality.

**Comment by Maleclypse** [comment]
> I'm not sure what point you're trying to make, felt has been made for thousands of years without the use of bubblewrap or any modern tools or casserole. Wool clumps can be easily separated by hand without cutting. The entire process is just applying a bit of pressure to condense wool fibers. You can do this with your own hands, water and soap in a manner of minutes just by using solid floor.
> 
> You can make an argument that it would be faster with certain tools, though a casserole serves absolutely no purpose in any of the guides you posted besides retaining the water so don't dirty your surroundings. Neither does a cutting implement make sense to create a patch, you shape the wool clumps with hands. Cutting comes when you're making clothing out of it, you don't cut your rags into precise shapes after you knit them.
> 
> It seems you're trying to artificially increase difficulty of this particular recipe without any basis in reality.
> 
> For the record, the recipe is using 200g of wool which seems reasonable for 15 minutes it takes. It would probably not dry that fast after being pressed into felt, but I'm not sure a wet mechanic is used for anything but towels.
> 
> > I love that your link uses a baking pan which is what a casserole dish represents in this recipe as well.
> 
> Do you understand the purpose of the baking pan? Please read above.

The point is keeping it in the time limit of 15 minutes.  I've asked you to go ahead and make the edits to get it to a hard floor, which isn't required in the current recipe.   Currently you can make felt patches on a shrubbery for all the game cares.  I can agree with you on the cutting of the patches happening elsewhere but it's not happening anywhere on many felt patch recipes in the game currently.

File: data/json/recipes/other/materials.json [hunk 2]
```diff
@@ -209,7 +209,9 @@
     "difficulty": 1,
     "time": "15 m",
     "autolearn": true,
-    "components": [ [ [ "soap", 5 ] ], [ [ "water_clean", 3 ] ], [ [ "wool_staple", 1 ] ] ]
+    "qualities": [ { "id": "CUT", "level": 1 } ],
```
Comment (Maleclypse) [inline data/json/recipes/other/materials.json:212]:
```suggestion

```

**Comment by Tommy-os** [comment]
> I've asked you to go ahead and make the edits to get it to a hard floor, which isn't required in the current recipe.

To my knowledge, there is only the "clean surface" quality which is used only for butchering and not for recipes. There is a "flat surface" (FLAT_SURF) property of terrain and furniture which is used for detecting dining furniture and deploying furniture on terrain, but not for crafting recipes.

> Currently you can make felt patches on a shrubbery for all the game cares.

That is the case for practically every crafting recipe in the game as far as I know. There's an impact on crafting speed if you do not have a proper workbench to craft on already.

> it's not happening anywhere on many felt patch recipes in the game currently.

If there are clothing recipes that do not require cutting of used textiles, that's most likely an issue with those recipes and not with crafting of the required components, which are quite generic in shape.

I see you've kept bubblewrap and casserole as required tools, with bubblewrap even somehow being consumed in the process. I'll leave it up to someone else to explain it to you as I've clearly failed at it.

**Comment by Maleclypse** [comment]
OK I finally found something worthwhile.   https://www.youtube.com/watch?v=S7gRIzcgDLQ   This video of a project took twelve hours and I understand that it's a finished project versus a patch but I'll try and guesstimate the area of this and if the patches recipe looks reasonable in retrospect of that I'll agree with you.  But the bubblewrap was standing in for the cheese cloth type stuff I keep seeing in any project that doesn't use bubblewrap which seems to be the modern standard.  I've got other stuff to do so I'll come back to this tonight.

**Comment by Tommy-os** [comment]
> modern standard

If you wish to include modern standards for production methods, I recommend doing it as new recipes instead of removing the primitive methods of achieving the same results. Many recipes already have alternatives for this very reason.

File: data/json/recipes/other/materials.json [hunk 3]
```diff
@@ -209,7 +209,8 @@
     "difficulty": 1,
     "time": "15 m",
     "autolearn": true,
-    "components": [ [ [ "soap", 5 ] ], [ [ "water_clean", 3 ] ], [ [ "wool_staple", 1 ] ] ]
+    "tools": [ [ [ "bubblewrap", -1 ] ], [ [ "casserole", -1 ] ] ],
+    "components": [ [ [ "soap", 5 ] ], [ [ "water_clean", 3 ] ], [ [ "wool_staple", 1 ] ], [ [ "bubblewrap", 1 ] ] ]
```
Comment (Maleclypse) [inline data/json/recipes/other/materials.json:213]:
```suggestion
    "components": [ [ [ "soap", 5 ] ], [ [ "water_clean", 3 ] ], [ [ "wool_staple", 1 ] ] ]
```

**Pull request merged**